src/common/providers/firestore.ts

```diff
@@ -52,9 +52,9 @@ function _getValueProto(data: any, resource: string, valueFieldName: string) {
 }
 
 /** @internal */
-export function createSnapshotFromProtobuf(data: Uint8Array, path: string) {
+export function createSnapshotFromProtobuf(data: Uint8Array, path: string, databaseId: string) {
   if (!firestoreInstance) {
-    firestoreInstance = firestore.getFirestore(getApp());
+    firestoreInstance = firestore.getFirestore(databaseId);
```

Comment:
You mean conditioning on `databaseId` right?

Because these two *fromProtobuf functions are only called from the V2 handler and V2 events always supply the `databaseId`